File: Arrays_Hashing/Encode_Decode/Encode_Decode.py
from typing import List
class Codec:        
    def encode(self, strs: List[str]) -> str:
        """Encodes a list of strings to a single string.
        """
        # Encode by 5/Hello5/World
        result = ""
        for word in strs:
            result += str(len(word)) + "/" + word
        return result

    def decode(self, s: str) -> List[str]:
        """Decodes a single string to a list of strings.
        """
        result = []
        i = 0
        while i < len(s):
            slash_index = s.index("/", i)
            length = int(s[i:slash_index])
            i = length+slash_index+1
            result.append(s[slash_index+1:i])
        return result



# Your Codec object will be instantiated and called as such:
# codec = Codec()
# codec.decode(codec.encode(strs))

File: Arrays_Hashing/Encode_Decode/Encode_decode.java
package Arrays_Hashing;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

public class Encode_decode {
    // Encodes a list of strings to a single string.
    public String encode(List<String> strs) {
        // add hashes to each of the strings to encode their values
        StringBuilder sb = new StringBuilder();
        for(String s: strs){
            sb.append(s.length()).append("/").append(s);
        }
        return sb.toString();
    }

    // Decodes a single string to a list of strings.
    public List<String> decode(String s) {
        List<String> result = new ArrayList<>();
        int i = 0;
        while(i < s.length()){
            int slash = s.indexOf("/", i);
            int size = Integer.parseInt(s.substring(i, slash));
            i = size+slash+1;
            result.add(s.substring(slash+1, i));
        }
        
        return result;
    }

    public static void main(String[] args) {
        String[] test = {"Hello","World"};
        List<String> strs = Arrays.asList(test);
        Encode_decode codec = new Encode_decode();
        System.out.println(codec.decode(codec.encode(strs)));

    }
}
